Hai Reveal: Korath background mission (#8075)
endless-sky/endless-sky: data/korath/korath missions.txt

# You should have received a copy of the GNU General Public License along with
# this program. If not, see <https://www.gnu.org/licenses/>.

# This mission sets the "know who the korath are" variable in old saves.

> mission "000 Do I know the Korath?"

Reviewer: Ok, but why does it need to go first?
All this does is encourage people to just add more and more `0`s, or whatever other character, to the beginning of their mission names, demonstrated by the fact that you've now got three here while some other things use only two.

> mission "000 Do I know the Korath?"

Author: Now it only has two. Its "on offer" needs to be applied before the conversation in "Korath Stormhold Artifacts [2] Drones"

	landing
	source
	to offer
		or
			has "event: label korath space"

> 			has "FW Pug 2B: offered"

Author: The purpose of the condition is to indicate that you know the Korath, and there must be a condition with that specific purpose so that missions can trigger on whether you know the Korath. Whether this comes from an event or a mission does not affect the necessity of a condition with that purpose.

	on complete
		payment 1000
		dialog `You find a Korath who seems to have been waiting for you in the spaceport. They greet you by showing their palms, then outstretch their hands as if waiting to receive the package. You hand off the small delivery and they hand you a small payment of <payment>.`

mission "Korath Stormhold Artifacts [1] Coded Message"
	name "Go to <planet>"
	description `Head to <planet> to deal with a "security issue."`
	landing
	to offer
		has "event: stormhold korath issue"

> 		not "Hai Reveal [B12] Devil-Hide Battle: accepted"

Reviewer 2: does accepted work? shouldnt we use offered instead?

Author: Done.

	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent" "Hai" "Hai (Unfettered)" "Pirate"
		not planet Stormhold
	destination Stormhold
	on offer
		conversation
			`As soon as you land, your communications console beeps urgently. It's a coded message from <planet> that simply reads, "Security issue. Return to <planet> at earliest convenience."`

> 			`	The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`

Reviewer 2: burning acorn? I like it
mb earliest convenience is not rushing enough though, and should be replaced by smth like "as soon as possible"?

Author: Done.

				accept

mission "Korath Stormhold Artifacts [2] Drones"

> 	name "Destroy Korath drones at <waypoints>."
> 	description `Destroy Korath stealth drones at <waypoints>. This is an optional mission; the Syndicate will do it if you don't.`
> 	landing

Reviewer 2: ```suggestion
	landing
	name "Destroy Korath drones at <waypoints>."
	description `Destroy Korath stealth drones at <waypoints>. This is an optional mission; the Syndicate will do it if you don't.`
```
landing tag should be above the name and description, by convention

Author: Applied manually.

	source Stormhold
	destination Stormhold
	waypoint Almach
	to offer
		has "Korath Stormhold Artifacts [1] Coded Message: done"
		not "Hai Reveal [B12] Devil-Hide Battle: accepted"
	on offer
		log "Factions" "Korath" `According to the Unfettered, the Korath were intelligent and well-cultured reptiles, until their civilization fell apart a few centuries ago. They peacefully coexisted with the Hai until they got desperate for resources and invaded. The Hai valiantly defended themselves against the Korath, but eventually the Korath nearly wiped themselves out through infighting. The few survivors live east of Syndicate space and frequently raid nearby human settlements.`
		log `There was an abundance of Korath artifacts at Stormhold, but Batu fears Stormhold was merely a staging area and sales distribution site. He expects this is part of a larger Korath outfit sales operation that is based on a nearby planet.`
		conversation

> 			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We've found many spoils of war on this planet. Sadly, there weren't any jump drives," he says, winking. "Fortunately, the pirates had plenty of other gadgetry littering their bunkers, including Hai and Korath outfits. We're gathering it to secure areas for inventory and analysis before divvying it up."`

Reviewer 2: ```suggestion
			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We have found many spoils of war on this planet. Sadly, there weren't any jump drives," he says, winking. "Fortunately, the pirates had plenty of other gadgetry littering their bunkers, including Hai and Korath outfits. We are gathering it to secure areas for inventory and analysis before divvying it up."`
```
I dont think Hai should use contractions, especially not an unfettered

Author: Applied manually.

				to display
					"reputation: Hai (Unfettered)" >= 30

> 			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We've found many spoils of war on this planet. The pirates' bunkers have had all manner of gadgetry, including Hai and Korath outfits. We're gathering it to secure areas for inventory and analysis before divvying it up."`

Reviewer 2: ```suggestion
			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We have found many spoils of war on this planet. The pirates' bunkers have had all manner of gadgetry, including Hai and Korath outfits. We are gathering it to secure areas for inventory and analysis before divvying it up."`
```
same

Author: Applied manually.

				to display
					"reputation: Hai (Unfettered)" < 30
			`	Batu looks less satisfied. "We've found far less than we expected, and there aren't any live Korath - or even Korath bodies. Our investigators suspect we've found a small part of a larger operation to pillage and sell Korath outfits. The operation's Stormhold facilities were just distribution centers for sales and staging areas for transfers of Korath. I hope shutting down this part of the operation will end it for good, but that seems overly optimistic."`

			# This seemingly abrupt question is to ensure the player notices the word "Korath."
			choice

> 				`	"Korath? Who are the Korath?"`

Reviewer 2: check Korath reputation, if its not -1000 we're sure the player has met them?

Author: The player is allowed to be forgetful.

Author: I changed it to, "Who are the Korath?" so the player can interpret it as asking Batu and Choot'k for their point of view (which is exactly what you're doing).

				`	"Why are there so many Korath artifacts?"`
					goto somany
			`	Choot'k responds. "The Korath are group of lizards that lived south of Hai space until few centuries ago. They used to be obsessed with art, music, and other esoterics, until they got desperate for resources and invaded us. Their technology is far beyond ours, and they used it to nearly wipe themselves out in bloody infighting. We learned a lot from them: even the most aggressive Unfettered are wary of attacking Hai planets, lest our species destroy itself like the Korath did. In all our raids in the Wah Ki system, never once have the Unfettered targeted civilian areas on Cloudfire." Batu appears shocked by this.`
			label somany

> 			`	Choot'k continues. "The most advanced group of surviving Korath lives in systems near yours, according to pirate data. It seems the Korath have been raiding ships in pirate and Syndicate systems for food, while the pirates have been disabling and pillaging Korath ships for tech. I'm not sure why the Korath have limited themselves to piracy. They certainly have the firepower to dominate a few dozen of your planets.`

Reviewer 2: if we took the second question he does not truly continue?

Author: The answer to the second question begins at that paragraph.

Author: Actually, I don't know what you're requesting. I may have misunderstood your question. Could you please clarify?

			`	"From our race's records, it is quite unlike the Korath to be so close to a more primitive race without making demands. Their resources may be committed to another war, or they may be building up supplies for a shock and awe style attack. It's anyone's guess at this point.`
			`	"Even the Unfettered are concerned about this. A few dozen human planets under the dominion of the Korath could make them a force to be reckoned with again. They could even be a threat to the Hai."`
			choice
				`	"Sounds foreboding."`
					goto foreboding
				`	"When do I get my cut of these spoils?"`
			`	"Ha! I'd think you were a True Hai if it weren't for the missing fur. Don't worry, captain, I'll make sure the others give you a suitable payment."`

> 			`	Batu sighs as Choot'k continues. "We've made an unpleasant discovery recently. The Korath have stealth drones hiding in two pirate systems: Alcyone and Almach. They've been using these to spy on nearby traffic, including us. It's also... eh, Batu?"`

Reviewer 2: stealth drones? I dont think they're really stealthed?

Author: They're stealthy enough so tier 1 races couldn't find them. It may not be part of the standard design, but these are stealthy.

Reviewer 2: yeah I know I just think we shouldnt use that word here but another one, idk, discrete, camouflaged, hidden,... but not stealthed because that has a specific meaning in the game

Author: The game is not presenting the word "stealth" as a fact; rather, it is the interpretation of the human and Hai investigators, which their leaders are parroting as the absolute truth. If you can't find drones for years, and they've been designed specifically, so you can't find them, then they're stealth drones from your point of view. Part of the player's job is to separate the real truth from the speaker's interpretation of the truth.

Reviewer 3: I'm not sure "stealth" really has that much of a specific meaning in-game, does it? It doesn't really have much meaning for me, at least. We have visible ships, and cloaked ships. So stealth is pretty much just a lore thing, not something that is in use in-game.

I'm good with these drones being called "stealth"

				goto end

			# Do you want a capital ship that can outrun drones?
			label afterburners
			action
				outfit "Afterburner (Planetary Class)"
				outfit "Fuel Processor" 3
			`	"Done! We're also giving you <payment>. Make sure you have enough cooling and engine space to support this afterburner."`
				goto end

			# This is a secret for players to find. It is the only time
			# in the game where you can get Firestorm Torpedo Racks.
			# Anything else you can pillage, so by some measure this is
			# the "right" option.
			label torpedoes
			action
				outfit "Firestorm Torpedo Rack" 2
				outfit "Firestorm Torpedo" 39
				outfit "Firestorm Battery"
			`	"Done! We're also giving you <payment>. Be careful with these torpedoes. They have a wide blast radius. They'll cripple your ship and escorts if they detonate too close."`
				goto end

			label end
			`	"It's always a pleasure to help another warrior get their due."`

	npc kill
		government Korath
		system Almach
		personality staying timid target marked mute
		fleet "Korath Stealth Relay"
		dialog "With all of the probes destroyed, you can now return to Stormhold for your payment."
	npc
		government Korath
		system Almach

> 		personality staying heroic marked mute opportunistic unconstrained

Reviewer 2: why mute?

Author: Not mute anymore.

		fleet
			variant

> 				"Ra'gru Ik 618"

Reviewer 2: I didnt just ask to make it capturable, I meant a capturable and fully crewed ship (look at the Ground Assault variant that already exists for another Korath ship)

Author: There is now a `Ra'gru Ik 618 (Ground Assault)` variant and it is used for the mission.

Author: The default 618 already has enough rifles for all of its bunks.

Reviewer 2: great, this small change should make it a bit harder to capture this guarantied to spawn ship

			label somany
			`	Choot'k continues. "The most advanced group of surviving Korath lives in systems near yours, according to pirate data. It seems the Korath have been raiding ships in pirate and Syndicate systems for food, while the pirates have been pillaging Korath ships for tech. I'm not sure why the Korath have limited themselves to piracy. They certainly have the firepower to dominate a few dozen of your planets.`
			`	"From our race's records, it is quite unlike the Korath to be so close to a more primitive race without making demands. Their resources may be committed to another war, or they may be building up supplies for a shock and awe style attack. It's anyone's guess at this point.`
			`	"Even the Unfettered are concerned about this. A few dozen human planets under the dominion of the Korath could make them a force to be reckoned with again. They could even be a threat to the Hai."`
			choice
				`	"Sounds foreboding."`
					goto foreboding
				`	"When do I get my cut of the spoils?"`

> 			`	"Ha! I'd think you were a True Hai if it weren't for the missing fur. Don't worry, captain, I'll make sure the others give you a suitable payment."`

Reviewer 4: ```suggestion
			`	"Ha! I'd think you were a True Hai if it weren't for the missing fur. Don't worry, Captain, I'll make sure the others give you a suitable payment."`
```

Author: Applied manually.

			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed computer algorithms that'll let you find them. Essentially, your computer finds where they aren't, and works inward from there." He looks up from his datapad at last.`
			`	Choot'k says, "We've destroyed the ones in this system, but we need you to fly to Almach and destroy the drones there."`
			choice
				`	"Just blow up drones? Why do you need me for this?"`
					goto whyme
				`	"I'm surprised you don't want the fight."`
				`	"Sounds like target practice to me."`
					goto practice

> 			`	"My troops need me here to command the pacification of Stormhold. I wish I could come with; it will be a glorious fight." He almost looks sad as he says that. "The largest ship we have seen since the Korath war now guards the drones."`

Reviewer 4: ```suggestion
			`	"My troops need me here to command the pacification of Stormhold. I wish I could come with; it will be a glorious fight." He almost looks sad as he says that. "The largest ship we have seen outside their space since the Korath war now guards the drones."`
```
I'd add something like this here, just as a hint that those JDs are being put to multiple uses. Will also go nicely with UHai campaign stuff to already have a reference to them occasionally keeping tabs on Korath space themselves.

Author: Even if they had the jump drives to spare for that, I doubt they would admit they're spying so close to the humans. Choot'k would be sharing valuable intelligence information that may reveal their sources. He's smart enough not to do that, especially when he's near a naval officer.

				goto navy
			label practice

> 			`	Choot'k looks worried. "You may be target practice if you aren't prepared. The drones are guarded now, by the largest ship we've seen since the Korath war. You'll need to dodge it while shooting the drones."`

Reviewer 4: ```suggestion
			`	Choot'k looks worried. "You may be target practice if you aren't prepared. The drones are guarded now, by the largest ship we've seen outside their space since the Korath war. You'll need to dodge it while shooting the drones."`
```

				goto navy
			label whyme

> 			`	Choot'k says, "Ah, it'll take a skilled pilot to do it. The drones are guarded now, by the largest ship we have seen since the Korath war. It is not attacking anything; we think it is just guarding the drones."`

Reviewer 4: ```suggestion
			`	Choot'k says, "Ah, it'll take a skilled pilot to do it. The drones are guarded now, by the largest ship we have seen outside their space since the Korath war. It is not attacking anything; we think it is just guarding the drones."`
```

				goto navy
			label navy
			`	Batu appears annoyed briefly before returning to his usual calm. "The Navy has prepared an elite strike force that is ready to take on the task. Teeneep has generously loaned us some Hai outfits to ensure our success. However, General Choot'k-"`
			`	Choot'k interrupts. "The urgency of my call was for you to have the right of first refusal. When you return here, successful in your mission, we'll be sure to have some good Korath gear ready for you."`
			`	Batu sighs. "The other leaders have agreed to this. However, if you would rather not do it, cancel the mission and the Navy will take care of it-" Choot'k cuts in, "-but don't expect to gain any respect for failure."`
			choice
				`	"Sound's like fun! I'm in."`
					accept
				`	"I'm not afraid, but I am busy right now."`
					decline
				`	"No thanks. I don't want to steal the Navy's glory."`
					decline
	on enter Almach
		dialog "The Navy's sensor algorithms are working: your ship is picking up an abundance of drones."
	on complete
		payment 2500000
		conversation

> 			`Choot'k greets you, beside some Navy crewman in inventory management uniforms. "The Navy tells me you shut down the transmissions. Good job! As promised, we've set aside some of the best Korath gear for you to choose from.`

Reviewer 4: ```suggestion
			`Choot'k greets you, beside some Navy crewmen in inventory management uniforms. "The Navy tells me you shut down the transmissions. Good job! As promised, we've set aside some of the best Korath gear for you to choose from.`
```
To match "uniforms," as just making that single seemed more awkward.

Author: Applied manually.

			`	"Pick among these:`
			`	"Some wonderful Korath systems that repair your hull and shields, while also storing energy. They come with two Korath anti-missile turrets and some other odds and ends.`

> 			`	"I can also offer the strongest afterburners I've ever heard of, plus a sizable amount of fuel storage and ramscoop to keep it working.`

Reviewer 4: ```suggestion
			`	"I can also offer the strongest afterburners I've ever heard of, plus a sizable amount of fuel storage and ramscoop to keep them working.`
```

> 			`	"I can also offer the strongest afterburners I've ever heard of, plus a sizable amount of fuel storage and ramscoop to keep it working.`

Reviewer 4: ```suggestion
			`	"I can also offer the strongest afterburner I've ever heard of, plus a sizable amount of fuel storage and ramscoop to keep it working.`
```
Or this one.

> 			`	"We can give you a power generator that's as big a fighter, and a lot of cooling outfits to support it.`

Reviewer 4: ```suggestion
			`	"We can give you a power generator that's as big as a fighter and a lot of cooling outfits to support it.`
```

> 			`	"We can give you a power generator that's as big as two fighters, and a lot of cooling outfits to support it.`

Reviewer 4: ```suggestion
			`	"We can give you a power generator that's as big as two fighters and a lot of cooling outfits to support it.`
```

Author: Applied manually while fixing another error on this line.

			`	"The Korath created a gigantic fiery turret they consider the pinnacle of their modern weapon design; you can have two of those.`
			`	"The last option is some torpedoes so devastating that even the Korath are afraid to use them. They're almost doomsday devices."`
			choice
				`	"Improved defenses sounds best."`
					goto defenses
				`	"Afterburners and fuel, of course."`
					goto afterburners
				`	"Power and cooling would be wonderful."`
					goto coolpower
				`	"I'll take the pair of fiery turrets."`
					goto inferno
				`	"Doomsday torpedoes? Count me in!"`
					goto torpedoes

			# This is the default option because you can't go wrong with defenses.
			# A mix of outfit types and sizes lets you outfit a wide variety of
			# designs. It's easy to pillage any of these, so this is the wimpiest
			# option, but also the most versatile.
			label defenses
			action
				outfit "Systems Core (Medium)"
				outfit "Systems Core (Small)"
				outfit "Warder Anti-Missile" 2
				outfit "Small Heat Shunt"
				outfit "Large Heat Shunt"
				outfit "Fuel Processor"

> 			`	"Done! We're also giving you <payment>. These outfits keep you alive through tough fights."`

Reviewer 4: ```suggestion
			`	"Done! We're also giving you <payment>. These outfits will keep you alive through tough fights."`
```

Author: Applied manually.

	on complete
		payment 1000
		dialog `You find a Korath who seems to have been waiting for you in the spaceport. They greet you by showing their palms, then outstretch their hands as if waiting to receive the package. You hand off the small delivery and they hand you a small payment of <payment>.`

mission "Korath Stormhold Artifacts [1] Coded Message"
	name "Go to <planet>"
	description `Head to <planet> to deal with a "security issue."`
	landing
	to offer
		has "event: stormhold korath issue"
		not "Hai Reveal [B12] Devil-Hide Battle: offered"

> 	source

Author: Doesn't the `source...not planet Stormhold` already take care of that?

```bash
        source
                government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent" "Hai" "Hai (Unfettered)" "Pirate"
                not planet Stormhold
```

> 	source

Reviewer 2: Oh sorry I'm on phone must have missed it

> 	source

Author: Yes, I just tried, and `source...not planet Stormhold` does prevent the mission from starting with a source of Stormhold.

		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent" "Hai" "Hai (Unfettered)" "Pirate"
		not planet Stormhold
	destination Stormhold
	on offer
		conversation
			`As soon as you land, your communications console beeps urgently. It's a coded message from <planet> that simply reads, "Security issue. Return to <planet> as soon as possible."`

> 			`	The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`

Reviewer 5: I'm not sure about the burning acorn because it's a species from Earth and I can't see why the Unfettered would use it when they might have a better symbol that seems more 'Hai' to me. But Hurleveur says they like it :#

Author: The frequent use of acorns is a running gag in Hai missions.

Reviewer 3: It makes as much sense as the Hai naming their ships after Earth creatures: Shield beetles, cicadas, sea scorpions, etc. I just assume that these are linguistic translations, and that the Hai have something that looks like an acorn, and they have species that are sufficiently similar in appearance and/or name to human equivalents that translators just translate them directly. 

It's probably giving biologists fits "No, no, that's not the seed from a *Quercus rubra*, but rather the seed from a Hai tree that we call the *Hai-Home Mercadus Flora Acornus.* Just because it *looks* like an acorn doesn't mean it *is* one."

Reviewer 3: It'd be really amusing to discover that whatever the Hai look-alike to an Acorn has some larger cultural relevance. For instance, perhaps it has some unusual medical properties: It completely suppresses pain and fatigue while inducing a state of high-activity. The sort of thing that you'd give to soldiers or rescue personnel. So eating human acorns has this weird sort of "I look like a tough person eating this thing that usually only people in physically demanding and life-threatening jobs eat, but without the negative of being wired and impossible to sleep for the next day."

Or perhaps it is the inverse: Something with intense calming properties that give Hai a strong "everything is fine, no need to worry" mental state for several hours. Perhaps used in religious ceremonies or something. So Hai enjoy eating human acorns because they are tasty and satisfying the cultural/religious push to consume these things; while avoiding the "mellowed out for several hours" after effect. (Having a significant part of the population routinely self-administering calming substances could be a major contributor to the long-term stability of the Hai)

On the flip side, maybe Hai acorns are just generically commonplace things, much like, say, wheat, rice, or maybe honey for humans. We use many agricultural symbols based on some of these in non-agricultural contexts. And it's the Earth originating acorns that have some decidedly unusual properties that the Hai are *very* fond of. It'd be amusing if that which is a regular nut for humans has strong pharmaceutical properties for the Hai that some like to use recreationaly. Perhaps it is a mild but significant effect, so they don't even recognize it. They just know that they *really* like those acorns from Earth... 

Lots of possibilities of what, exactly, the acorn could represent to the Hai people in general, and to the organizations that use it in specific. Also a lot of amusing and/or meaningful cultural reasons as to why human acorns are so popular.

Reviewer 5: > It makes as much sense as the Hai naming their ships after Earth creatures: Shield beetles, cicadas, sea scorpions, etc. I just assume that these are linguistic translations, and that the Hai have something that looks like an acorn, and they have species that are sufficiently similar in appearance and/or name to human equivalents that translators just translate them directly.
> 
> It's probably giving biologists fits "No, no, that's not the seed from a _Quercus rubra_, but rather the seed from a Hai tree that we call the _Hai-Home Mercadus Flora Acornus._ Just because it _looks_ like an acorn doesn't mean it _is_ one."

Well, that was what I thought the names of the Hai ships were like, too. I only pointed out the acorn because I had thought the Unfettered were around before regular humanity met the Hai, and it just didn't seem right that they'd use a foreign symbol for something that seems to be a nationalistic movement. And yes, it would be much better if these ideas were actually found in game, that would be an interesting conversation for the player.

Author: All add a hybrid of what Zitchas and bene-dictator said in my revisions to A11. The revised version will give you a second route through the dialog: read intelligence reports instead of attending the council proceedings, and then get a summary at the end that'll make you wish you were at the proceedings. (This will make more sense when you see the PR.)

I'll add an "intelligence report" about the remarkable similarities between species, and how common biological pressures lead to common evolution of species on different planets. Also, I'll research the physiological effects of various acorn species on squirrels, so I can have plausible technobabble.

Reviewer 5: I don't think a report of the similarities of Human acorns and Hai acorns is very relevant to this Korath mission. I would have thought that it might be more interesting to be it's own mission outside of any campaign; maybe a human scientist wants to study them after Hai Revealed, for example, to create a unique blend that would be commercially successful or to improve medicinal use or whatever. If it's included in this mission, I'd expect people to ignore it so they can get onto the action of destroying Korath faster instead.

Author: What? No, not this mission. The super-long Elder Council proceedings that took MasterOfGrey six weeks to write.

Reviewer 5: Yeah, I know, I didn't read it properly.

> 			`	The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`

Reviewer: Someone gave the Unfettered private keys for digital signatures that appear to humans as `diplomatic`?
I wonder who else they're sending these messages to.
Wasn't there a big deal about a communications blackout earlier? What happened to that?

Reviewer 3: it's entirely possible that the Unfettered just have their own diplomats and signatures that label themselves as diplomatic. This does just say that the signature claims to be diplomatic, not that it matches authentic diplomatic codes issued elsewhere. In the real world, vast numbers of scammers send out emails claiming to be diplomatic staff on a regular basis. Every once in a while they even fool the people who actually deal with diplomats on a regular basis and should know how to verify that.

Chances are they aren't super concerned about a coms blackout, either.

Reviewer: A "digital signature" is a thing irl. If jargon like this is to be used, it ought to be used accurately.
It is not as simple as forging a signature. A scammer sending out an email is unlikely to be able to produce a digital signature that genuinely appears to be from a legitimate source unless they have compromised that legitimate source and acquired their private key, and done so without being detected (else the digital signature produced by that private key would be invalidated).
When checking a digital signature for authenticity, a verified list of digital certificates is used.

> it's entirely possible that the Unfettered just have their own diplomats and signatures that label themselves as diplomatic.

If this were the case, then in order to verify their digital signature, we would have to use their own security certificates, at which point it should be pretty obvious: "The digital signature matches some certificates provided by the Hai; the message is from the Unfettered." Though, I feel this would be an unnecessary level of detail.

> This does just say that the signature claims to be diplomatic, not that it matches authentic diplomatic codes issued elsewhere.

See my earlier comments on what a digital signature.
Here are some other, probably better, explanations:
https://www.adobe.com/sign/digital-signatures.html#:~:text=A%20digital%20signature%20is%20a,document%20and%20can%20be%20verified.
https://support.microsoft.com/en-us/office/digital-signatures-and-certificates-8186cd15-e7ac-4a16-8597-22bd163e8e96
https://www.docusign.co.uk/how-it-works/electronic-signature/digital-signature/digital-signature-faq

> In the real world, vast numbers of scammers send out emails claiming to be diplomatic staff on a regular basis. Every once in a while they even fool the people who actually deal with diplomats on a regular basis and should know how to verify that.

Again, such situations either do not make use of legitimate digital signatures or do so using stolen credentials.
Have the Unfettered stolen these credentials? And they're using them to send *me* a message? And leaving a symbol representing themselves in it? Doesn't really make sense.

> Chances are they aren't super concerned about a coms blackout, either.

Everything here indicates that they are using hte normal human infrastructure to distribute this message. The Hai might not care about hte blackout, but the people in control of that infrastructure do.

Reviewer 3: Thank you for the detailed explanation. That being said, it is *very* common for high security organizations to use cryptographic signatures that do not have a public key distributed outside of their organization, generally specifically because if it's important enough to warrant it, they don't want it leaving their network in the first place. Or they distribute the appropriate key to the recipient ahead of time.

In any case, I don't see a problem here because of the wording:

> The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`

It **claims** to be of diplomatic origin. The description doesn't say that it successfully passes any kind of verification. The fact that it ends with "it is most likely from the unfettered." also adds to the element of doubt

I've seen plenty of documents that state that they are digitally signed, but then got printed out and scanned at some point; or just refer to an internal company cryptographic source that my software, being outside the company, can't access. So I can see that it **claims** to be from XYZ corporation, but that it cannot be verified. Yes, both these situations render the digital signature absolutely and completely useless in terms of security or guaranteeing origin. But it still tells me who it *claims* to be from, though.

Lastly, there's no indication of *whose* diplomatic group it originated from, other than the logo. That *also* reinforces the idea that the "diplomatic digital signature" is *not* verified or confirmed, since if it *was* verified, we would know for sure whose diplomatic corps it came from.

And as an aside, while I know that "Digital Signatures" are indeed a formalized thing, I've met an awful lot of people, organizations, and even companies that consider the signature line at the bottom of their email to be their digital signature. They typed their name and info into it, that is, to them, as good as putting pen to paper. Given how much of a hassle it is to verify a signature (not to mention how easy they are to duplicate and forge; and the utter lack of standards or protocols dealing with verifying signatures), to most people the two *are* functionally just as good as each other.

Reviewer: > it is very common for high security organizations to use cryptographic signatures that do not have a public key distributed outside of their organization, generally specifically because if it's important enough to warrant it, they don't want it leaving their network in the first place.

If this is the case here, and the player character does not have access to the public key for this digital signature, it is not possible to verify it.

> Or they distribute the appropriate key to the recipient ahead of time.

If this were the case here, the player character would know already who it came from. Either the Unfettered gave it to us to ensure their communications can be verified (in which case, there is no real need for the acorn symbol) or one of the government or military officials involved in this operation gave it to us, in which case they would also presumably tell us it is for the Unfettered, instead of just saying it is "diplomatic" (which would allow any of the individuals with the private key to impersonate any diplomat we think could reasonably send this message.)

> The description doesn't say that it successfully passes any kind of verification.

If it hasn't, that's a fairly important detail to leave out. This narrator is really not very good.

> I've seen plenty of documents that state that they are digitally signed, but then got printed out and scanned at some point; or just refer to an internal company cryptographic source that my software, being outside the company, can't access. So I can see that it claims to be from XYZ corporation, but that it cannot be verified. Yes, both these situations render the digital signature absolutely and completely useless in terms of security or guaranteeing origin. But it still tells me who it claims to be from, though.
> 
> Lastly, there's no indication of whose diplomatic group it originated from, other than the logo. That also reinforces the idea that the "diplomatic digital signature" is not verified or confirmed, since if it was verified, we would know for sure whose diplomatic corps it came from.

Again, we should really be told if the digital signature is determined to be fake, or can't be verified.
Not doing so and just saying "Well, I didn't say it was verified," when it is pointed out that it doesn't really make sense just seems like lazy writing.

> And as an aside, while I know that "Digital Signatures" are indeed a formalized thing, I've met an awful lot of people, organizations, and even companies that consider the signature line at the bottom of their email to be their digital signature. They typed their name and info into it, that is, to them, as good as putting pen to paper. Given how much of a hassle it is to verify a signature (not to mention how easy they are to duplicate and forge; and the utter lack of standards or protocols dealing with verifying signatures), to most people the two are functionally just as good as each other.

Here, we are being told this by the *narrator*. Not a character that probably doesn't understand the jargon themselves.
If the narrator says it is a digital signature, it should be a digital signature, not what people who don't actually know what it means think it means.
More generally, if the narrator is going to use technical jargon, it should be used in a technically correct manner.

Reviewer 3: I suspect that most players can understand the nuance behind "claimed," if they chose to actually think about it. I don't think we need to spell it out for them. In fact, I'd generally say that forcing content creators to hit people in the face with "this is not verified, you do not know who this is from" blunt statements is actually a fairly strong negative to our writing.

That being said, swapping "digital" before "signature" for "block" afterwards should be a reasonable change that removes the reference to the technical definition without having to degrade the text with unnecessary bluntness.

```suggestion
			`	The signature block claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`
```

Reviewer: I don't think a digital signature is an appropriate thing to insert nuances like "It might not be from who it says it is," because the whole point of a digital signature is that it is verifiable, easily so in a digitally connected world.

Not using such technical terminology is indeed an improvement, but I would just say `signature`, not `signature block`, what does hte `block` mean here? Are you just trying to link this to a digital signature without using that specific term?

If you want to talk about player's picking up on nuance, though, the thing calling out the Unfettered as the probable source isn't really necessary.

Maybe just drop the idea of a "diplomatic signature" altogether. They clearly already have a calling card.

```suggestion
			`	It is signed with a burning acorn insignia.`
```

Reviewer 2: I havent read all of it but to give context, my idea is that in general the Unfettered dont send info over coms, at least not critical one. Hence why its nice the msg itself doesnt say much.
They wouldnt have much experience writing this kind of stuff, I imagine it was done by Hai sympathizers that came with the fleet and had all clearance needed to make it a Hai certified thing if needed, or they made it on their own but it'd be pretty rough

Author: I removed the "likely from the Unfettered" part.

Author: > I haven't read all of it but to give context, my idea is that in general the Unfettered dont send info over coms, at least not critical one. Hence why its nice the msg itself doesnt say much.

This isn't a critical mission; it is Choot'k giving you right of first refusal on an exciting fight because he can't do the mission, and wants to live vicariously through you.

Reviewer 3: > I don't think a digital signature is an appropriate thing to insert nuances like "It might not be from who it says it is," because the whole point of a digital signature is that it is verifiable, easily so in a digitally connected world.
> 

And yet, verifying someone is who they say they are is increasingly difficult, to the point I sometimes have trouble convincing people I am who I am.

> Not using such technical terminology is indeed an improvement, but I would just say `signature`, not `signature block`, what does hte `block` mean here? Are you just trying to link this to a digital signature without using that specific term?

I swapped a term that you objected to, as it is a technical term and it was being used in the not-functioning-properly form instead of working properly, for another technical term that is in widespread use and probably will continue to be used for as long as emails and text communications are a thing. See https://en.wikipedia.org/wiki/Signature_block . They often have pictures too, unlike a digital signature, which tends not to.

Reviewer: > And yet, verifying someone is who they say they are is increasingly difficult, to the point I sometimes have trouble convincing people I am who I am.

You're not using digital signatures, though.

Author: We should not assume the distant future will still be digital.

Reviewer 6: How about IFF? I only skimmed this discussion though

Reviewer: > We should not assume the distant future will still be digital.

Then don't use modern day technical jargon.

Author: > Then don't use modern day technical jargon.

Any message that has to authoritatively state that it comes from a specific source will have a "signature" or "seal" of some sort. That is a tradition going back thousands of years, and there is no reason to believe it won't exist in a few thousand from now.

It is important to the dialog that Captain `<last>` knows the message is signed by an unknown diplomatic authority with a flaming acorn insignia. If someone can come up with a better way to state that, without removing any of that information, then I am open to changing it.

Reviewer 7: Let's imagine you get this as an email. It is digitally signed. You don't have the public key, so you import it from a keyserver. Now it's verified that the sender created the private key, but not the identity of the sender. You check the email address associated with it and it says "diplomat@gmail.com." That is a digital signature that claims to be of diplomatic origin, but you have no idea if the person whose email is "diplomat@gmail.com" and who created the private key is an actual diplomat.

Author: The player has the freedom to ignore the suspicious message, if they want to.

				accept

mission "Korath Stormhold Artifacts [2] Drones"
	landing
	name "Destroy Korath drones at <waypoints>."

> 	description `Destroy Korath stealth drones at <waypoints>. This is an optional mission; an elite Navy group will do it if you don't.`

Reviewer 5: ```suggestion
	description `Destroy Korath stealth drones at <waypoints>. This is an optional mission; an elite Navy group will do it if you choose not to.`
```

Author: The navy will also finish the mission if you die. Your choice is less important to them than your success.

	source Stormhold
	destination Stormhold
	waypoint Almach
	to offer
		has "Korath Stormhold Artifacts [1] Coded Message: done"
		not "Hai Reveal [B12] Devil-Hide Battle: offered"
	on offer
		log "Factions" "Korath" `According to the Unfettered, the Korath were intelligent and well-cultured reptiles, until their civilization fell apart a few centuries ago. They peacefully coexisted with the Hai until they got desperate for resources and invaded. The Hai valiantly defended themselves against the Korath, but eventually the Korath nearly wiped themselves out through infighting. The few survivors live east of Syndicate space and frequently raid nearby human settlements.`
		log `There was an abundance of Korath artifacts at Stormhold, but Batu fears Stormhold was merely a staging area and sales distribution site. He expects this is part of a larger Korath outfit sales operation that is based on a nearby planet.`
		conversation

> 			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We found many spoils of war on this planet. Sadly, there weren't any jump drives," he says, winking. "The pirates had plenty of other gadgetry littering their bunkers, including limited-access human goods and Hai equipment. We've found a few Korath outfits, too, including some quite advanced ones. Navy inventory crewmen are gathering it to secure areas for inventory and analysis before divvying it up."`

Reviewer 3: ```suggestion
			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We found many spoils of war on this planet. Sadly, there weren't any jump drives," he says, winking. "The pirates had plenty of other gadgetry littering their bunkers, including limited-access human goods and Hai equipment. We've found a few Korath outfits, too, including some quite advanced ones. It looks like these pirates must have gotten lucky and managed to actually loot one of their ships! Navy inventory crewmen are gathering it to secure areas for inventory and analysis before divvying it up."`
```

Just a small extra sentence to convey that pirates looting a Korath ship is a really impressive and rare thing; and to establish that this is all potentially just from a single ship. This is in part to counter the player's perception of things vs the storytelling detail. A Palavret carries up to 250 people, with 777 tons of outfits, and the ship itself is another 720. That's a huge amount of stuff. But a player can look at a Palavret and go "Eh, that's only 21 outfits (not counting rifles or ammunition). That's not much stuff. So if this base has *lots* of stuff, they must have captured lots of ships!" Not realizing that the story is looking at *all* the artifacts coming off a ship, not just the big individual outfits that the player cares about. Counting all the personal effects and things that are items but not outfits, a Palavret probably has tens of thousands of parts. 

(For comparison, the average car has about 30k parts; and if you count all the nuts and bolts and things like those as just being part of larger items, the average car still has about 1800 individual parts. So something like the Palavret... Yeah, probably actually in the millions of parts...

				to display
					"reputation: Hai (Unfettered)" >= 30

> 			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We found many spoils of war on this planet. The pirates' bunkers had plenty of gadgetry littering their bunkers, including limited-access human goods and Hai equipment. We've found a few Korath outfits, too, including some quite advanced ones. Navy inventory crewmen are gathering it to secure areas for inventory and analysis before divvying it up."`

Reviewer 3: ```suggestion
			`Choot'k greets you, with a worried-looking Batu at his side. "Welcome back, Captain <last>! We found many spoils of war on this planet. The pirates' bunkers had plenty of gadgetry littering their bunkers, including limited-access human goods and Hai equipment. We've found a few Korath outfits, too, including some quite advanced ones. It looks like these pirates must have gotten lucky and managed to actually loot one of their ships! Navy inventory crewmen are gathering it to secure areas for inventory and analysis before divvying it up."`
```

			label questions
			choice
				`	"Who are the Korath?"`
					to display
						not "know who the korath are"
						"reputation: Korath" > -100
						not "asked choot'k about the korath"
				`	"I've fought the Korath for a long time, but I don't really know who they are."`
					to display
						not "know who the korath are"
						"reputation: Korath" <= -100
						not "asked choot'k about the korath"
				`	"What can you tell me about the Korath?"`
					to display
						not "asked choot'k about the korath"
						or
							has "know who the korath are"
							"reputation: Korath" <= -100
				`	"Why are there so many Korath artifacts?"`
					goto somany
			action
				set "know who the korath are"
				set "asked choot'k about the korath"

> 			`	Choot'k responds. "The Korath are a group of lizards that lived south of Hai space until a few centuries ago. They used to be obsessed with art, music, and other esoterics. I care more for practical matters, but I must admit that I enjoy our old recordings of their music. Haunting guttural hums accented by angelic vocals with lyrics in their intricate language. The mood goes from eerie and hopeful to enlivening to frightening from verse to verse and draws the listener with them. Accompanied by mesmerizing visuals, the experience absorbs you in alluring totality."`

Reviewer 5: Would Choot'k call them lizards, or a lizard-like species? Is it meant to sound like he doesn't like them?

Author: Referring back to the [Korath lore document](https://docs.google.com/document/d/1875bt5lLh4GH_TE_9EsoAvwwk0DPJnLkVt-ztTvfv1g/edit#), they are a "species of partially-feathered, hyper-adaptable, omnivorous *lizard*." He is merely stating a fact. It's like saying that humans are hairless bipeds. To someone who has never seen a human, it is an invaluable piece of information.

Author: If he had said, "feather-munching, nigh-cannibalistic, apocalypse-obsessed, cockroach-fearing, dancing, art freaks," then *that* would be insulting, albeit essentially true.

Reviewer 2: well in a way calling them lizards is like calling us apes and I would prefer if he didnt have a grudge against them (for the Unfettered campaign)
so good you specified that

Author: No, there would be no grudge. The Hai and Korath were friends for two millennia, and the Hai had to watch helplessly as the Korath destroyed themselves. This wasn't ancient history, either, because there are Hai still alive from that time. They would be in mourning, not enraged.

Author: Indeed, the views of the people who actually fought the Korath (Unfettered Hai) would be different from the ones watching the suffering from afar (Fettered Hai).

> 			`	Choot'k responds. "The Korath are partially-feathered, omnivorous, sentient, lizards that lived south of Hai space until a few centuries ago. They used to be obsessed with art, music, and other esoterics. I care more for practical matters, but I must admit that I enjoy our old recordings of their music. Haunting guttural hums accented by angelic vocals with lyrics in their intricate language. The mood goes from eerie and hopeful to enlivening to frightening from verse to verse and draws the listener with them. Accompanied by mesmerizing visuals, the experience absorbs you in alluring totality."`

Reviewer 5: ```suggestion
			`	Choot'k responds. "The Korath are partially-feathered, omnivorous, sentient lizards that lived south of Hai space until a few centuries ago. They used to be obsessed with art, music, and other esoterics. I care more for practical matters, but I must admit that I enjoy our old recordings of their music. Haunting guttural hums accented by angelic vocals with lyrics in their intricate language. The mood goes from eerie and hopeful to enlivening to frightening from verse to verse and draws the listener with them. Accompanied by mesmerizing visuals, the experience absorbs you in alluring totality."`
```
Sorry, just this.

> 			`	He looks slightly embarrassed and switches to practical matters. "The Hai and Korath were were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`

Reviewer 4: ```suggestion
			`	He looks slightly embarrassed and switches to practical matters. "The Hai and Korath were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`
```

Author: I don't know how I keep missing these things.

> 			`	He looks slightly embarrassed and switches to practical matters. "The Hai and Korath were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`

Reviewer 5: ```suggestion
			`	He looks slightly embarrassed and switches back to the subject. "The Hai and Korath were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`
```

Author: This odd wording refers back to the previous paragraph where he claims he, "cares more for practical matters," while implying that may not be 100% true.

> 			`	He looks slightly embarrassed and switches to practical matters. "The Hai and Korath were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`

Reviewer 5: ```suggestion
			`	He looks slightly embarrassed and switches to the practical matters. "The Hai and Korath were on good terms - trading goods and culture - until the lizards got desperate for resources and invaded us. Their technology is far beyond ours, but we defended valiantly. I'm not sure we would have won on our own, but they nearly wiped themselves out in bloody infighting. Ultimately, it was stronger and older races that ended a self-destructive spiral towards extinction. As an Unfettered, I have grievances with the Drak, but I am glad those space cockroaches saved the Korath from themselves.`
```
I felt it was repetitive before. But if you want it that way.

			`	"From our race's records, it is quite unlike the Korath to be so close to a more primitive race without making demands. Their resources may be committed to another war, or they may be building up supplies for a shock and awe style attack. It's anyone's guess at this point.`
			`	"Even the Unfettered are concerned about this. A few dozen human planets under the dominion of the Korath could make them a force to be reckoned with again. They could even be a threat to the Hai."`
			choice
				`	"Sounds foreboding."`
					goto foreboding
				`	"When do I get my cut of the spoils?"`
			`	"Ha! I'd think you were a True Hai if it weren't for the missing fur. Don't worry, Captain, I'll make sure the others give you a suitable payment."`
			`	Batu sighs as Choot'k continues. "We've made an unpleasant discovery recently. The Korath have stealth drones hiding in two pirate systems: Alcyone and Almach. They've been using these to spy on nearby traffic, including us. It's also... eh, Batu?"`
				goto next
			label foreboding

> 			`	"Indeed it does, especially with our recent discovery. The Korath have stealth drones hiding in two pirate systems: Alcyone and Almach. They've been using these to spy on nearby traffic, including us. It's also... eh, Batu?"`

Reviewer 4: "It's also... eh, Batu?"

We could also expand this to something more like "It's also... eh, what was the technical explanation again, Batu?"

Author: I tweaked it a bit. Do you like it now?

			label next

> 			`	Batu nods and pulls up something on his datapad. He reads mechanically to you. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Author: The length of the text, the use of technical jargon, and Batu's confusion, emphasizes how advanced the tactic was that the Korath were using. I'm impressing upon readers the brilliance of the Korath's methods, and how much planning went into it. If the reader doesn't understand the jargon, then it'll feel like the Korath have done crazy technical magic, which is the intended effect. The reader does not need to understand the text, or even do more than skim over it, to achieve the desired effect.

On the off chance that a player knows what a phase array sounder and transceiver is, they'll still be impressed that the Korath made one that is nearly impossible to detect, out of drones that are constantly moving.

> 			`	Batu nods and pulls up something on his datapad. He reads mechanically to you. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Author: By the way, what is going through Batu's head, is similar to what is going through yours. ("Why did Navy Intelligence have to explain this to me? I feel like they could have given me just the practical info. It would have been less of a text wall") That's why he is "reading mechanically," and he's only doing it because Choot'k wanted him to relay the information. Most likely, the main take-away Choot'k got from the long rambling explanation is what I pointed out: Korath made stealth drones using a brilliant technical feat that is beyond us.

> 			`	Batu nods and pulls up something on his datapad. He reads mechanically to you. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Reviewer 2: Yeah I suspected this was smth like that but just so you know I'd skip over the conversation at that jargon, expecting it to not really end
Still, I'm not expert at content creating ofc

> 			`	Batu nods and pulls up something on his datapad. He reads mechanically to you. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Author: I shall await feedback from others on this subject. Does anyone have an opinion on whether I should make my point another way?

> 			`	Batu nods and pulls up something on his datapad. He reads mechanically to you. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Reviewer 4: ```suggestion
			`	Batu nods and pulls up something on his datapad, which he begins to recite mechanically. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`
```

That came through for me, but if we need to make it clearer...

> 			`	Batu nods and pulls up something on his datapad, which he recites mechanically. "The drone fleet acts as a phase array transceiver and hyperwave sounder. A phase array uses signals from multiple sources (the drones) transmitting or receiving slightly out of phase, to emulate a much larger antenna. It's the same principle as phase array radars used for meteorology and phase array telescopes for astronomy. Each drone's communication system is not much stronger than a human hyperspace relay, but the fleet acts together as one powerful relay. We think it's able to relay signals as far as 120 light-years, without the need for starlanes. What's most concerning is that relaying is only a secondary functionality; they're mostly observation systems.`

Author: It works on the same principle as jump drives. The weft (hyperspace) is everywhere, regardless of the presence of starlanes. Primitive races can only send signals, or ships, into the weft through starlanes. The Korath know how to send signals (and ships) without a starlane.

The stealth comes from the same principle. Humans can only triangulate interstellar hyperspace signals that come through starlanes. Without that aid, any interstellar signal simply baffles them.

> 			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed computer algorithms that'll let you find them. Essentially, your computer finds where they aren't, and works inward from there." He looks up from his datapad at last.`

Reviewer 4: ```suggestion
			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed computer algorithms that'll let you find them. Essentially, your computer finds where it can be sure they aren't and works through a process of elimination from there." He looks up from his datapad at last.`
```
Something like this, making it more explicit could help with the explaining meaningfully as well, I think.

Author: I reworked those two sentences, and included a game developer inside joke. The algorithm uses a quad tree, which is a data structure for searching 2D space. You'd need an octree to search 3D space, so this is an inside joke about combat in Endless Sky being 2D.

> 			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed computer algorithms that'll let you find them. Essentially, your computer finds where they aren't, and works inward from there." He looks up from his datapad at last.`

Reviewer 4: ```suggestion
			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed computer algorithms that'll let you find them. Essentially, your computer finds where they aren't and works inward from there." He looks up from his datapad at last.`
```
But also, spare comma I missed earlier should go either way. :)

> 			`	He slides his finger across the pad and continues. "They were hard to detect, and have probably been hiding for years, perhaps decades. From debris analysis, we determined the drones are painted with a material that tricks human sensors, so seeing them is tough. Their transmissions are cleverly disguised as ambient noise, so you can't listen for them either. However, we've developed a quad tree search algorithm to find the probes. The algorithm detects everywhere the probes cannot be, and all that's left are probe locations." He looks up from his datapad at last.`

Author: The Unfettered are now helping the Hai. Long ago, they were at war with the Korath, and they haven't forgotten the old Korath tricks. As for why they were looking, they found evidence of surveillance when they were searching Stormhold. Of course, the Unfettered also knew the Korath Exiles were nearby, and would have taken advantage of the situation to monitor the Exiles.

			`	Choot'k says, "We've destroyed the ones in this system, but we need you to fly to Almach and destroy the drones there."`
			choice
				`	"Just blow up drones? Why do you need me for this?"`
					goto whyme
				`	"I'm surprised you don't want the fight."`
				`	"Sounds like target practice to me."`
					goto practice

> 			`	"My troops need me here to command the pacification of Stormhold. I wish I could come with; it will be a glorious fight." He almost looks sad as he says that. "The largest ship we have seen since the Korath war now guards the drones."`

Author: The Korath saw the attack at Stormhold, and know that the drones above Stormhold are gone. Since they're monitoring the humans, they may have already realized the ones at Almach were discovered. Even if they don't, it's obvious that multinational taskforce who can detect and destroy Stormhold drones, will soon detect and destroy the others. The Korath want to send a message that destroying their property is unwise. After the deed is done, they'll develop better stealth drones and send those instead.

				goto navy
			label practice
			`	Choot'k looks worried. "You may be target practice if you aren't prepared. The drones are guarded now, by the largest ship we've seen since the Korath war. You'll need to dodge it while shooting the drones."`
				goto navy
			label whyme
			`	Choot'k says, "Ah, it'll take a skilled pilot to do it. The drones are guarded now, by the largest ship we have seen since the Korath war. It is not attacking anything; we think it is just guarding the drones."`
				goto navy
			label navy
			`	Batu appears annoyed briefly before returning to his usual calm. "The Navy has prepared an elite strike force that is ready to take on the task. Teeneep has generously loaned us some Hai outfits to ensure our success. However, General Choot'k-"`

> 			`	Choot'k interrupts. "The urgency of my call was for you to have the right of first refusal. When you return here, successful in your mission, we'll be sure to have some good Korath gear ready for you."`

Author: I wanted it to be an approximate translation from an Unfettered cultural tradition that has no direct translation in the human language. Quotes would make that more clear.

> 			`	Choot'k interrupts. "The urgency of my call was for you to have the right of first refusal. When you return here, successful in your mission, we'll be sure to have some good Korath gear ready for you."`

Reviewer 2: Quotes sound good to make that clearer indeed

			`	Batu sighs. "The other leaders have agreed to this. However, if you would rather not do it, cancel the mission and the Navy will take care of it-" Choot'k cuts in, "-but don't expect to gain any respect for failure."`
			choice

> 				`	"Sound's like fun! I'm in."`

Reviewer 5: ```suggestion
				`	"Sounds like fun! I'm in."`
```

			label afterburner
			action
				outfit "Afterburner (Planetary Class)"
				outfit "Fuel Processor" 3
			`	"Done! We're also giving you <payment>. Make sure you have enough cooling and engine space to support this afterburner."`
				goto end

			# This is a secret for players to find. It is the only time
			# in the game where you can get Firestorm Torpedo Racks.
			# Anything else you can pillage, so by some measure this is
			# the "right" option.
			label torpedoes
			action
				outfit "Firestorm Torpedo Rack" 2
				outfit "Firestorm Torpedo" 39
				outfit "Firestorm Battery"
			`	"Done! We're also giving you <payment>. Be careful with these torpedoes. They have a wide blast radius. They'll cripple your ship and escorts if they detonate too close."`
				goto end

			label end
			`	"It is always a pleasure to help another warrior get their due."`

	npc kill
		government Korath
		system Almach
		personality staying timid target marked
		fleet "Korath Stealth Relay"
		dialog "With the probes destroyed, you can now return to Stormhold for your payment."
	npc
		government Korath
		system Almach
		personality staying heroic marked opportunistic
		fleet
			variant

> 				"Ra'gru Ik 618 (Ground Assault)"

Reviewer 3: A "Ground Assault" variant doesn't seem like the most appropriate name for a ship on guard duty to protect some drones. Is there a reason why the Korath would use a ship calibrated for Ground Assault missions as a sentry here?

As per the comments on the ship definition, I'm suggesting using a more appropriate (Sentry) label here instead of the (Ground Assault), since the latter suggests a ship that has different outfitting for a ground invasion, whereas the former simply states what it is:, a Sentry.

```suggestion
				"Ra'gru Ik 618 (Sentry)"
```

> 		fleet "Korath Stealth Relay"

Reviewer: Why not just define the fleet in here?

Author: Done.

> 		personality staying heroic marked opportunistic

Reviewer: Why is this marked?

Author: The ship is only there to kill whoever attacks the drones. It isn't a general raiding vessel; it has a specific mission. Others are wise enough to stay out of the way.

	on complete
		payment 1000
		dialog `You find a Korath who seems to have been waiting for you in the spaceport. They greet you by showing their palms, then outstretch their hands as if waiting to receive the package. You hand off the small delivery and they hand you a small payment of <payment>.`

mission "Korath Stormhold Artifacts [1] Coded Message"
	landing

> 	name "Go to <planet>"

Author: Done.
